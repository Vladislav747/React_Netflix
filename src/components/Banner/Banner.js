import React, { useState, useEffect } from "react";

import "./Banner.css";
import requests from "../../requests";
import axios from "../../axios";

function Banner({ fetchUrl }) {
  const [movie, setMovies] = useState([]);

  const base_url = "https://image.tmdb.org/t/p/original/";

  //run once when the row loads, and don't run again
  useEffect(() => {
    async function fetchData() {
      const request = await axios.get(requests.fetchNetflixOriginals);
      setMovies(
        request.data.results[
          Math.floor(Math.random() * request.data.results.length - 1)
        ]
      );
      return request;
    }
    fetchData();
  }, []);

  console.log(movie, "moviesBanner");
  return (
    <header
      className="banner"
      style={{
        backgroundImage: `url("https://image.tmdb.org/t/p/original/${movie.backdrop_path}")`,
      }}
    >
      <div className="banner__contents">
        <h1>{movie?.title || movie.name || movie.original_name}</h1>
      </div>
    </header>
  );
}

export default Banner;
