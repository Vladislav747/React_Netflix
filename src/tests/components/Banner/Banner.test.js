import renderer from "react-test-renderer";
import React from "react";

import Banner from "../../../components/Banner/Banner";

describe("Render Banner", () => {
  it("render correctly Banner component", () => {
    const rowComponent = renderer.create(<Banner />).toJSON();
    expect(rowComponent).toMatchSnapshot();
  });

  it("has class all div elements", () => {
    const component = shallow(<Banner />);
    expect(component.find("header.banner")).toHaveLength(1);
    expect(component.find(".banner__contents")).toHaveLength(1);
    expect(component.find(".banner__description")).toHaveLength(1);
    expect(component.find(".banner__button")).toHaveLength(2);
  });
});
