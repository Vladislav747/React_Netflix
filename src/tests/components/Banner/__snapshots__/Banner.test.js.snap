// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Banner render correctly Banner component 1`] = `
<header
  className="banner"
  style={
    Object {
      "backgroundImage": "url(\\"https://image.tmdb.org/t/p/original/undefined\\")",
    }
  }
>
  <div
    className="banner__contents"
  >
    <h1 />
    <div
      className="banner__buttons"
    >
      <button
        className="banner__button"
      >
        Play
      </button>
      <button
        className="banner__button"
      >
        Play
      </button>
    </div>
    <h1
      className="banner__description"
    />
  </div>
  <div
    className="banner--fadeBottom"
  />
</header>
`;
